### PR TITLE
Multiple user sessions supported now

### PR DESCRIPTION
**Pull request by aashishpanta0**

### CODE REVIEW

File: src/openvisuspy/slice.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def onIdle(self):
 
 		# I need to fix the aspect ratio 
 		if self.pdim==2 and [self.last_W,self.last_H]!=[W,H]:
-			x+=0.5*w
+			x+=0.5*w # changing x+=0.5*w to x only, same for y+=0.5*h
```
Comment (scrgiorgio) [inline src/openvisuspy/slice.py:102]: what changed here?

File: src/openvisuspy/slice.py [hunk 2]
```diff
@@ -269,64 +270,57 @@ def showData(self, data, viewport,color_bar=None):
 
 # ////////////////////////////////////////////////////////////////////////////////////
 class Slice(param.Parameterized):
+	def __init__(self): # just so that we can get new instances in each session
```
Comment (scrgiorgio) [inline src/openvisuspy/slice.py:275]: Are you sure you have to do this? I thought the `param` library was automatically creating instance sessions automatically. We need to dig into it.

File: src/openvisuspy/slice.py [hunk 3]
```diff
@@ -855,25 +938,24 @@ def setSceneBody(self, scene):
 			directions = {it: I for I, it in enumerate(directions)} if directions else  {'X':0,'Y':1,'Z':2}
 		self.direction.options=directions
 
-		self.timestep_delta.value=int(scene.get("timestep-delta", 1))
+		self.timestep_delta.value=int(scene.get("timestep-delta", 64))
```
Comment (scrgiorgio) [inline src/openvisuspy/slice.py]: what if the dataset has 1 timestep? this is the most common case...

File: src/openvisuspy/slice.py [hunk 4]
```diff
@@ -855,25 +938,24 @@ def setSceneBody(self, scene):
...
 		self.view_dependent.value = bool(scene.get('view-dependent', True))
 
 		resolution=int(scene.get("resolution", -6))
 		if resolution<0: resolution=self.db.getMaxResolution()+resolution
 		self.resolution.end = self.db.getMaxResolution()
 		self.resolution.value = resolution
-
 		self.field.value=scene.get("field", self.db.getField().name)
-		self.num_refinements.value=int(scene.get("num-refinements", 1 if pdim==1 else 2))
+		self.num_refinements.value=int(scene.get("num-refinements", 1))
```
Comment (scrgiorgio) [inline src/openvisuspy/slice.py]: I am scared by interactivity setting `1` by default. At least with 2 you have a way to get the  first coarse level and move/abort query quickly...

File: src/openvisuspy/slice.py [hunk 5]
```diff
@@ -1219,8 +1302,8 @@ def gotNewData(self, result):
 
 			# in dynamic mode, I need to use the data range
 			if mode=="dynamic":
-				self.range_min.value = data_range[0]
-				self.range_max.value = data_range[1]
+				self.range_min.value = round(data_range[0],6) # I am trying to avoid too many refreshes
```
Comment (scrgiorgio) [inline src/openvisuspy/slice.py]: It seems reasonable, but what for datasets which are quickly moving in some spot in a range <1e-6? Not sure if this is too much dataset-dependent.

**Review by scrgiorgio — COMMENTED** [review]
Very good overall

**Comment by scrgiorgio** [comment]
@aashishpanta0

File: src/openvisuspy/slice.py (hunk 3 above)
Comment (aashishpanta0) [reply src/openvisuspy/slice.py]: right, this was for a different usecase. will send a new push

File: src/openvisuspy/slice.py (hunk 2 above)
Comment (aashishpanta0) [reply src/openvisuspy/slice.py:275]: It was not creating new instances earlier, most likely sharing the session states. Tried fixing the main.py file, and it wasnot working..  atleast I didnot find any other ways to do it except this.

File: src/openvisuspy/slice.py (hunk 4 above)
Comment (aashishpanta0) [reply src/openvisuspy/slice.py]: OKay, will dial back

**Comment by aashishpanta0** [comment]
@scrgiorgio fixed

**Review by aashishpanta0 — COMMENTED** [review]
Looks good now

**Pull request merged**